src/EnergyPlus/RoomAirModelManager.cc

```diff
@@ -2692,233 +2715,204 @@ namespace RoomAir {
-                EquipFind = true;
-                break;
-            }
-        }
-
-        ErrorObjectHeader eoh{routineName, equipTypeName, EquipName};
-
         switch (zoneEquipType) {
         case DataZoneEquipment::ZoneEquipType::VariableRefrigerantFlowTerminal: { // ZoneHVAC:TerminalUnit : VariableRefrigerantFlow
-            SupplyNodeName = Alphas(4);
-            ReturnNodeName = ""; // Zone return node
+            SupplyNodeNum = HVACVariableRefrigerantFlow::GetVRFTUZoneInletAirNode(state, EquipIndex);
         } break;
         case DataZoneEquipment::ZoneEquipType::EnergyRecoveryVentilator: { // ZoneHVAC : EnergyRecoveryVentilator
-            int fanIndex = Fans::GetFanIndex(state, Alphas(4));
-            if (fanIndex == 0) {
-                ShowSevereItemNotFound(state, eoh, "", Alphas(4));
-                errorfound = true;
-            }
-
-            int nodeNum = state.dataFans->fans(fanIndex)->outletNodeNum;
-            if (errorfound) {
-            }
-            SupplyNodeName = state.dataLoopNodes->NodeID(nodeNum); // ?????
-            ReturnNodeName = "";                                   // Zone exhaust node
+            SupplyNodeNum = HVACStandAloneERV::GetStandAloneERVOutAirNode(state, EquipIndex);
         } break;
         case DataZoneEquipment::ZoneEquipType::FourPipeFanCoil: { // ZoneHVAC : FourPipeFanCoil
-            SupplyNodeName = Alphas(6);
-            ReturnNodeName = Alphas(5);
+            SupplyNodeNum = FanCoilUnits::GetFanCoilZoneInletAirNode(state, EquipIndex, EquipName);
+            ReturnNodeNum = FanCoilUnits::GetFanCoilAirInNode(state, EquipIndex, EquipName);
         } break;
         case DataZoneEquipment::ZoneEquipType::OutdoorAirUnit: { // ZoneHVAC : OutdoorAirUnit
-            SupplyNodeName = Alphas(13);
-            ReturnNodeName = Alphas(14);
+            SupplyNodeNum = OutdoorAirUnit::GetOutdoorAirUnitZoneInletNode(state, EquipIndex);
+            ReturnNodeNum = OutdoorAirUnit::GetOutdoorAirUnitReturnAirNode(state, EquipIndex);
         } break;
         case DataZoneEquipment::ZoneEquipType::PackagedTerminalAirConditioner: { // ZoneHVAC : PackagedTerminalAirConditioner
-            SupplyNodeName = Alphas(4);
-            ReturnNodeName = Alphas(3);
+            SupplyNodeNum = state.dataUnitarySystems->unitarySys[EquipIndex - 1].getAirOutNode(state, EquipName, 0, errorfound);
+            ReturnNodeNum = state.dataUnitarySystems->unitarySys[EquipIndex - 1].getAirInNode(state, EquipName, 0, errorfound);
         } break;
         case DataZoneEquipment::ZoneEquipType::PackagedTerminalHeatPump: { // ZoneHVAC : PackagedTerminalHeatPump
-            SupplyNodeName = Alphas(4);
-            ReturnNodeName = Alphas(3);
+            SupplyNodeNum = state.dataUnitarySystems->unitarySys[EquipIndex - 1].getAirOutNode(state, EquipName, 0, errorfound);
+            ReturnNodeNum = state.dataUnitarySystems->unitarySys[EquipIndex - 1].getAirInNode(state, EquipName, 0, errorfound);
```

Comment:
@lgu1234 and I discussed using a getIndex function in the parent objects, specifically for all objects managed by UnitarySystem, to directly access the state variables for UnitarySystem. So this should look like:

    int usIndex = UnitarySys::GetIndex(state, EquipName);  <-- this will need EquipType if names are not unique
    SupplyNodeNum = state.dataUnitarySystems->unitarySys[usIndex].AirOutNode;
    ReturnNodeNum = state.dataUnitarySystems->unitarySys[usIndex].AirInNode;